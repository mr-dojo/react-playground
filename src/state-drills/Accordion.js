import React, { Component } from 'react';

export default class Accordion extends Component {
  static defaultProps = {
    sections: [] 
  };
  state = {
    currentSectionIndex: Number,
  };
  handleButtonClick = (index) => {
    this.setState({ currentSectionIndex: index})
  }
  renderSections() {
    return this.props.sections.map((section, index) => (
      <li>
        <button key={index} onClick={() => this.handleButtonClick(index)}>
          {section.title}
        </button>
        {!!this.props.sections.length || (this.state.currentSectionIndex >= 0 && this.renderContent(index))}
      </li>
      ))
    }  
  renderContent(index) {
    const currentSection = this.props.sections[index]
    return (
      <p className='content'>
        {currentSection.content}
      </p>
    )
  }
  render() { 
    return (
      <ul>
        {this.renderSections()}
      </ul>
    )
  }
}