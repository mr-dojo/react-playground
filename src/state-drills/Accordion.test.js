import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion.js';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

describe(`Accordion Component`, () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  it('renders without errors', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  
  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('renders the first tab as expected', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('closes the first tab and opens any clicked tab', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    console.log('>>> WRAPPER <<<')
    console.log(wrapper.debug())
    console.log('>>> FIND(BUTTON) <<<')
    console.log(wrapper.find('button').at(1).simulate('click'))
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
});