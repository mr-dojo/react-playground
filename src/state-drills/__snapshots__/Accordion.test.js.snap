// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first tab and opens any clicked tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <Component />
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <Component />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <Component />
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no tabs 1`] = `<ul />`;

exports[`Accordion Component renders the first tab as expected 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
    <Component />
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
    <Component />
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
    <Component />
  </li>
</ul>
`;
